index.bs

```diff
@@ -5523,6 +5529,126 @@ opened</dfn> steps given |window|, |type|, |message|, and optional |default valu
+
+The <dfn export for=modules>emulation</dfn> module contains commands and events
+relating to emulation of browser APIs.
+
+### Definition ### {#module-emulation-definition}
+
+[=remote end definition=]
+
+<pre class="cddl remote-cddl">
+EmulationCommand = (
+  emulation.setGeolocationOverride
+)
+</pre>
+
+A <dfn>geolocation override</dfn> is a [=struct=] with an [=struct/item=] named
+<dfn attribute for="geolocation override">latitude</dfn> which is a float,
+a [=struct/item=] named <dfn attribute for="viewport-dimensions">longitude</dfn> which is a float,
+a [=struct/item=] named <dfn attribute for="viewport-dimensions">accuracy</dfn> which is a float,
+a [=struct/item=] named <dfn attribute for="viewport-dimensions">altitude</dfn> which is a float or null,
+a [=struct/item=] named <dfn attribute for="viewport-dimensions">altitudeAccuracy</dfn> which is a float or null,
+a [=struct/item=] named <dfn attribute for="viewport-dimensions">heading</dfn> which is a float or null,
+a [=struct/item=] named <dfn attribute for="viewport-dimensions">speed</dfn> which is a float or null.
+
+A [=remote end=] has a <dfn>geolocation overrides map</dfn> which is a weak map between [=user contexts=] and [=geolocation override=].
+
+
+### Commands ### {#module-emulation-commands}
+
+#### The emulation.setGeolocationOverride Command ####  {#command-emulation-setGeolocationOverride}
+
+The <dfn export for=commands>emulation.setGeolocationOverride</dfn> command modifies geolocation characteristics on the given top-level traversables or user contexts.
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+    <pre class="cddl remote-cddl">
+      emulation.SetGeolocationOverride = (
+        method: "emulation.setGeolocationOverride",
+        params: emulation.SetGeolocationOverrideParameters
+      )
+
+      emulation.SetGeolocationOverrideParameters = {
+        coordinates: emulation.GeolocationCoordinates / null,
+        ? contexts: [+browsingContext.BrowsingContext],
+        ? userContexts: [+browser.UserContext],
+      }
+
+      emulation.GeolocationCoordinates = {
+         latitude: float,
+         longitude: float,
+         ? accuracy: float .default 1.0,
+         ? altitude: float / null .default null,
+         ? altitudeAccuracy: float / null .default null,
+         ? heading: float / null .default null,
+         ? speed: float / null .default null,
+      }
+    </pre>
+   </dd>
+   <dt>Result Type</dt>
+   <dd>
+    <code>
+     EmptyResult
+    </code>
+   </dd>
+</dl>
+
+<div algorithm="remote end steps for emulation.setGeolocationOverride">
+
+The [=remote end steps=] with |command parameters| are:
+
+1. If |command parameters| [=map/contains=] "<code>userContexts</code>"
+   and |command parameters| [=map/contains=] "<code>context</code>",
+   return [=error=] with [=error code=] [=invalid argument=].
+
+1. Let |navigables| be a [=/set=].
+
+1. If the <code>contexts</code> field of |command parameters| is present:
```

Comment:
We now have some helpers for parts of this function, like https://w3c.github.io/webdriver-bidi/#get-valid-navigables-by-ids. Given that a lot of these commands end up doing basically the same setup of "get a list of top-level-traversables, fail if any arguments aren't that" (and similar things for user contexts), maybe we can try to centralise more of the logic in those (or new, related) helpers.

Reply:
Sure, make sense, added two new helpers for top-level traversables and user contexts.